src/deploy/functions/release/fabricator.ts

```diff
@@ -0,0 +1,463 @@
+    }
+
+    const resultFunction = (await this.functionExecutor
+      .run(async () => {
+        const op: { name: string } = await gcfV2.updateFunction(apiFunction);
+        return await poller.pollOperation<gcfV2.CloudFunction>({
+          ...gcfV2PollerOptions,
+          pollerName: `update-${endpoint.region}-${endpoint.id}`,
+          operationResourceName: op.name,
+        });
+      })
+      .catch(rethrowAs(endpoint, "update"))) as gcfV2.CloudFunction;
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const serviceName = resultFunction.serviceConfig.service!;
+    if (backend.isHttpsTriggered(endpoint) && endpoint.httpsTrigger.invoker) {
+      await this.executor
+        .run(() =>
+          run.setInvokerUpdate(endpoint.project, serviceName, endpoint.httpsTrigger.invoker!)
+        )
+        .catch(rethrowAs(endpoint, "set invoker"));
+    }
+
+    if (endpoint.concurrency) {
+      await this.setConcurrency(endpoint, serviceName, endpoint.concurrency);
+    }
+  }
+
+  async deleteV1Function(endpoint: backend.Endpoint): Promise<void> {
+    const fnName = backend.functionName(endpoint);
+    await this.functionExecutor
+      .run(async () => {
+        const op: { name: string } = await gcf.deleteFunction(fnName);
+        const pollerOptions = {
+          ...gcfV1PollerOptions,
+          pollerName: `delete-${endpoint.region}-${endpoint.id}`,
+          operationResourceName: op.name,
+        };
+        await poller.pollOperation<void>(pollerOptions);
+      })
+      .catch(rethrowAs(endpoint, "delete"));
+  }
+
+  async deleteV2Function(endpoint: backend.Endpoint): Promise<void> {
+    const fnName = backend.functionName(endpoint);
+    await this.functionExecutor
+      .run(async () => {
+        const op: { name: string } = await gcfV2.deleteFunction(fnName);
+        const pollerOptions = {
+          ...gcfV2PollerOptions,
+          pollerName: `delete-${endpoint.region}-${endpoint.id}`,
+          operationResourceName: op.name,
+        };
+        await poller.pollOperation<void>(pollerOptions);
+      })
+      .catch(rethrowAs(endpoint, "delete"));
+  }
+
+  async setConcurrency(
+    endpoint: backend.Endpoint,
+    serviceName: string,
+    concurrency: number
+  ): Promise<void> {
+    await this.functionExecutor
+      .run(async () => {
+        const service = await run.getService(serviceName);
+        if (service.spec.template.spec.containerConcurrency === concurrency) {
+          logger.debug("Skipping setConcurrency on", serviceName, " because it already matches");
+          return;
+        }
+
+        delete service.status;
+        delete (service.spec.template.metadata as any).name;
+        service.spec.template.spec.containerConcurrency = concurrency;
+        await run.replaceService(serviceName, service);
+      })
+      .catch(rethrowAs(endpoint, "set concurrency"));
+  }
+
+  async setTrigger(endpoint: backend.Endpoint): Promise<void> {
```

Comment:
Is the plan here to add the work we need for future 'special' trigger types like task queues to this function? If so, I very much like the idea, but a brief comment explaining it would be useful.